smithed_libraries/plugins/versioning/load.py

```diff
@@ -0,0 +1,237 @@
+from contextlib import suppress
+
+from beet import Advancement, Context, Function, FunctionTag, Generator
+
+from .models import VersioningOptions
+from .utils import call_if_version_match
+
+
+def load_tags(
+    ctx: Context,
+    opts: VersioningOptions,
+    dependencies: str,
+    enumerate: str,
+    resolve: str,
+):
+    """Sets the Lantern Load tags for deps, enumeration, and resolution"""
+
+    load_tag = ctx.generate(
+        opts.lantern_load.tag_path,
+        merge=FunctionTag(
+            {
+                "values": [
+                    {"id": f"#{dependencies}", "required": False},
+                    f"#{enumerate}",
+                    f"#{resolve}",
+                ]
+            }
+        ),
+    )
+
+    with ctx.override(generate_namespace="load"):
+        ctx.generate(
+            opts.lantern_load.step, merge=FunctionTag({"values": [f"#{load_tag}"]})
+        )
```

Comment:
Since lantern load is supposed to have static, well-defined paths, it's a bit counterintuitive to make use of the context generator, whose role is primarily to facilitate working with dynamic paths. I would probably just make `load_tags()` return a `DataPack` instance containing the required resources and merge it into `ctx.data` later on.

Reply:
This is probably true but I'll leave it for now.